package exceptionLearn;

import java.util.Date;
import java.util.InputMismatchException;
import java.util.Scanner;


/*
 * 编译时异常和运行时异常
 * 
 * 包:java.lang.Throwable
 * 两个子类Error和Exception
 * 
 * 
 * */
public class errorTest {

	public static void main(String[] args) {
		// TODO Auto-generated method stub
//		main(args);  栈溢出
		
		
//		编译时异常
//		例如文件异常处理
		
		
//		运行时异常
//		NullPointerException
//		String str=null;
//		System.out.println(str.charAt(0));
		
//		ArrayIndexOutOfBoundsException
//		String str="ab";
//		System.out.println(str.charAt(6));
		
//		ClassCastException类型转换异常
//		Object obj=new Date();
//		String str=(String)obj;
		
//		NumberFormatException
//		String str="abc";
//		int num=Integer.parseInt(str);
		
//		InputMismatchException
//		Scanner scanf=new Scanner(System.in);
//		int num=scanf.nextInt();
//		System.out.println(num);
		
//		ArithmeticException
//		System.out.println(10/0);
		
		 
	}

}













